relevanceai/http_client.py

```diff
@@ -99,3 +101,102 @@ def check_auth(self):
         except:
             raise Exception("Invalid auth details.")
+
+    def kmeans_cluster(self,
+        dataset_id: str,
+        vector_fields: list,
+        filters: List = [],
+        k: Union[None, int] = 10,
+        init: str = "k-means++",
+        n_init: int = 10,
+        max_iter: int = 300,
+        tol: float = 1e-4,
+        verbose: bool = True,
+        random_state = None,
+        copy_x: bool = True,
+        algorithm: str ="auto",
+        alias: str = "default",
+        cluster_field: str="_cluster_",
+        overwrite: bool = False
+    ):
+        """
+        This function performs all the steps required for Kmeans clustering:
+        1- Loads the data
+        2- Clusters the data
+        3- Updates the data with clustering info
+        4- Adds the centroid to the hidden centroid collection
+
+        Parameters:
+        dataset_id : string
+            name of the dataser
+        vector_fields : list
+            a list containing the vector field to be used for clustering
+        filters : list
+            a list to filter documents of the dataset,
+        k : int
+            K in Kmeans
+        init : string
+            "k-means++" -> Kmeans algorithm parameter
+        n_init : int
+            number of reinitialization for the kmeans algorithm
+        max_iter : int
+            max iteration in the kmeans algorithm
+        tol : int
+            tol in the kmeans algorithm
+        verbose : bool
+            True by default
+        random_state = None
+            None by default -> Kmeans algorithm parameter
+        copy_x : bool
+            True bydefault
+        algorithm : string
+            "auto" by default
+        alias : string
+            "default", string to be used in naming of the field showing the clustering results
+        cluster_field: string
+            "_cluster_", string to name the main cluster field
+        overwrite : bool
+            False by default, To overwite an existing clusering result
+
+        """
+        if '.'.join([cluster_field, vector_fields[0], alias]) in self.datasets.schema(dataset_id) and not overwrite:
+            self.logger.error("Clustering results already exist.")
+            sys.exit(1)
+
+        # load the documents
+        docs = self.get_all_documents(dataset_id=dataset_id, filters=filters, select_fields=vector_fields)
+
+        # Cluster
+        clusterer = KMeans(
+            k = k,
+            init = init,
+            n_init = n_init,
+            max_iter= max_iter,
+            tol= tol,
+            verbose = verbose,
+            random_state = random_state,
+            copy_x = copy_x,
+            algorithm = algorithm)
+        clustered_docs = clusterer.fit_documents(vector_fields, docs, alias = alias, cluster_field = cluster_field, return_only_clusters=True)
+
+        # Updating the db
+        try:
+            results = self.update_documents(dataset_id, clustered_docs, chunksize = 50)
```

Comment:
Can you set the chunksize parameter to the top? This should be `update_documents_chunksize`